src/serde/de_br.rs

```diff
@@ -338,7 +340,43 @@ mod tests {
         );
 
         let mut list: Vec<(NodePtr, Option<NodePtr>)> = vec![(n1, None), (n2, None)];
+        let list_node_part = a.new_pair(n1, NodePtr::NIL).unwrap();
+        let list_node = a.new_pair(n2, list_node_part).unwrap();
 
+        assert_eq!(
+            traverse_path_with_vec(&mut a, &[0b10], &mut list).unwrap(),
+            n2
+        );
+
+        assert_eq!(
+            traverse_path_with_vec(&mut a, &[0b101], &mut list).unwrap(),
+            n1
+        );
+        assert_eq!(
+            traverse_path_with_vec(&mut a, &[0b111], &mut list).unwrap(),
+            nul
+        );
+        // confirm we have no cache before we call the function
+        assert!(list[0].1.is_none());
+        assert!(list[1].1.is_none());
+        a.add_ghost_pair(2).unwrap();
+        let test_val = traverse_path_with_vec(&mut a, &[0b1], &mut list).unwrap();
+        assert!(node_eq(&a, test_val, list_node));
+        // check cache has been created
+        assert!(list[0].1.is_some());
+        assert!(list[1].1.is_some());
```

Comment:
you could even compare these, with `node_eq()` against `list_node_part` and `list_node`, right?
They won't be the same nodes, but they should compare equal.

Reply:
done